src/components/Tooltip.js

```diff
@@ -0,0 +1,180 @@
+import _ from 'underscore';
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import {Animated, Text, View} from 'react-native';
+import Hoverable from './Hoverable';
+import withWindowDimensions, {windowDimensionsPropTypes} from './withWindowDimensions';
+import getTooltipStyles from '../styles/getTooltipStyles';
+
+const propTypes = {
+    // The text to display in the tooltip.
+    text: PropTypes.string.isRequired,
+
+    // Children to wrap with Tooltip.
+    children: PropTypes.node.isRequired,
+
+    // Props inherited from withWindowDimensions
+    ...windowDimensionsPropTypes,
+};
+
+class Tooltip extends Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            // The distance between the left side of the wrapper view and the left side of the window
+            xOffset: 0,
+
+            // The distance between the top of the wrapper view and the top of the window
+            yOffset: 0,
+
+            // The width and height of the wrapper view
+            wrapperWidth: 0,
+            wrapperHeight: 0,
+
+            // The width and height of the tooltip itself
+            tooltipWidth: 0,
+            tooltipHeight: 0,
+        };
+
+        // The wrapper view containing the wrapped content along with the Tooltip itself.
+        this.wrapperView = null;
+
+        // The tooltip (popover) itself.
+        this.tooltip = null;
+
+        this.animation = new Animated.Value(0);
+
+        this.getWrapperPosition = this.getWrapperPosition.bind(this);
+        this.measureWrapperAndGetPosition = this.measureWrapperAndGetPosition.bind(this);
+        this.measureTooltip = this.measureTooltip.bind(this);
+        this.showTooltip = this.showTooltip.bind(this);
+        this.hideTooltip = this.hideTooltip.bind(this);
+    }
+
+    componentDidUpdate(prevProps) {
+        if (!_.isMatch(this.props.windowDimensions, prevProps.windowDimensions)) {
```

Comment:
I think maybe this should be `isEqual`? Not really sure if `isMatch` does the same thing https://underscorejs.org/#isMatch